browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -48,6 +49,15 @@
 
 namespace brave {
 
+namespace {
+
+const std::string& GetCanonicalName(
+    const std::vector<std::string>& dns_aliases) {
+  return dns_aliases.size() >= 1 ? dns_aliases.front() : base::EmptyString();
```

Comment:
@antonok-edm

Reply:
Looks like this may as well return `absl::optional<std::string>` given how it's called, although this doesn't seem to change any behavior.

Reply:
Just kept it this way as that's how it was upstream before they removed this method, which I think has the advantage of you returning always the same actual piece of data (i.e. a const reference), instead of a new value each time. WDYT?

Reply:
No strong preference from me - but your logic here makes sense!